src/tiered/tiered_work.c

```diff
@@ -156,11 +156,15 @@ __wt_tiered_flush_work_wait(WT_SESSION_IMPL *session, uint32_t timeout)
             __wt_epoch(session, &now);
         }
         /* We are done if we don't find any work units or exceed the timeout. */
-        done = !found || (WT_TIMEDIFF_SEC(now, start) > timeout);
+        if (WT_TIMEDIFF_SEC(now, start) > timeout)
+            timed_out = true;
+        done = !found || timed_out;
     }
-    if (!found)
+    if (timed_out) {
+        WT_ASSERT(session, !found);
```

Comment:
We set `now` in the `if (found)` block, above. Then we use `now` to decide if we've exceeded the timeout. So won't this ASSERT always trigger?

Reply:
We initialize `now` before the loop to the current time. So if there is no work to do at all, `now` is set, but `found` won't be true but we will not have timed out. That being said, if the only thing we're doing is printing out an informational message but returning 0/success, is knowing the timeout hit worthwhile?

Maybe the right change is to just remove that message.